### DISCUSSION TITLE
Can we create workflows in multiple namespace based upon a sensor in another namespace?

### DISCUSSION THREAD

**Discussion by usamaB** [argoproj/argo-events#1620] [Q&A] [answered]
Hey, I have a use case and I’m not fully aware of the possible solutions.
So basically based upon an event(kafka mesasge) I want to Trigger Argo-workflows in multiple namespaces.
Each team has its own namespace and I want to separate stuff and just deploy a workflow in their namespace.

I can think of two solutions for the use-case:

1. Have a Service account with ClusterRole. Create multiple Argoworkflowtiggers in a sensor or multiple sensors in the same namespace where event-source and eventbus are. I have tested this, it works. But I don’t like this solution because I want to automate the pipeline, provide a helm chart to the teams. So they just only need to give the container image and their service account. And with this solution, it doesn’t seem feasible and nice as all is in one place.
2.  Is it possible for An EventSource to write to multiple EventBuses in different namespaces so the sensors in those namespaces can trigger?  If it is possible then Deploy EventBuss and Sensors In different namespaces so they can trigger a workflow in the respective namespace. Else deploying all 3 (eventtsource, eventtbus, sensor) is also not good as would have to deploy the same secrets in each namespace and then a consumer group for each. 

Are there any other ways or if these can be improved?

**Comment by usamaB** [comment, marked as answer]
Solved! I used the second method.

**Reply by katharina-schulze** [reply]
Hi @usamaB, can you explain how you achieved an EventSource writing to multiple EventBuses in different namespaces? I would like to do this too, but can't find a way.

**Reply by KrisF-Midnight** [reply]
> Hi @usamaB, can you explain how you achieved an EventSource writing to multiple EventBuses in different namespaces? I would like to do this too, but can't find a way.

I don't think it's possible. The EventSource API doesn't give you the option to specify the namespace for the EventBus. It only gives you option to specify the name. https://argoproj.github.io/argo-events/APIs/#argoproj.io/v1alpha1.EventSourceSpec
